### ISSUE TITLE
Unable to reproduce the data metrics on OKVQA - GS112K.

### ISSUE THREAD

**Issue by blcahk** [xinwei666/MMGenerativeIR#3]
Hi, thanks for sharing this project. I tried running the code following the provided instructions, but I couldn’t reproduce the reported metrics.

For the original data, I used the data under the ./dataset directory in the code and the COCO2014 dataset. Since I couldn't find the model weights decapoda-research/llama-7b-hf in the code, I used linhvu/decapoda-research-llama-7b-hf instead.

The regional features and FM index initialization were carried out according to the instructions you provided, resulting in feature and index files. Finally, the model weights were obtained by training on the A6000 with the original configuration. But the data obtained during the evaluation phase is far inferior to the metrics in your paper.

<img width="884" height="230" alt="Image" src="https://github.com/user-attachments/assets/3471eaed-4d4a-475c-8d47-56f7dde1f24f" />

After that, I used the model weights you provided, but the results I obtained are still not ideal.

<img width="874" height="125" alt="Image" src="https://github.com/user-attachments/assets/7822861a-298f-420b-b819-720fc152e7a1" />

Could you please clarify if there are additional settings (e.g., random seeds, environment versions, or hyperparameters) that I should pay attention to in order to reproduce the reported results? Any guidance or suggestions would be greatly appreciated. Thanks a lot! 🙏

**Comment by xinwei666** [comment]
Thank you for your interest. I am also puzzled by the results you reproduced. This work was completed two years ago, and we used the official LLAMA model. I am not sure whether you are using the same backbone as I did. I recommend that you use the same model as we did, along with the fm-index we provided.

Second, I noticed a warning in the screenshot you provided: "... right padding is detected." Please verify whether right padding is causing this issue. I suggest you **test it again using eval_batch_size=1**.

Third, please cross-check the environment I provided (requirement.txt) and confirm whether the LoRA weights have been loaded correctly, as well as other hyperparameters.

Fourth, please confirm whether you have modified the code in the eval script (such as hyperparameters for beam search) or the metric calculation methods (for example, the p@k and r@k in this paper refer to pseudo-precision and pseudo-recall, following previous work).

Thank you again for your attention. If you have any other clues, feel free to discuss further. Before uploading this repository, we re-ran the code and reproduced results that were slightly better than those reported in the paper. Additionally, this repository has been successfully replicated by other research teams.

**Comment by xinwei666** [comment]
We have checked the evaluation logs as shown below. 

![Image](https://github.com/user-attachments/assets/f1aad13b-aa46-433e-a30e-a1639c459638)

Additionally, we have located our retrieval results file (results_okvqa_20240611004_p5: 0.491.json). The numerical ID strings here correspond to the IDs and texts in our provided id2text.json.

**Comment by blcahk** [comment]
Thank you very much for your guidance. After adjusting the batch size to 1 in the evaluation script, I was able to successfully reproduce the metrics reported in the paper.

<img width="1255" height="260" alt="Image" src="https://github.com/user-attachments/assets/85043407-f8c4-4db6-929a-2a1ec51cb847" />
